### PR TITLE
#184759911 Implement Displaying Users

### PR DESCRIPTION
**Pull request by Munezeroolivierhugue**
## What does this PR do?

This PR implements displaying users feature

## Description of Tasks to be completed?

Admin should be able to:
- View all users
- Update roles of user
- change status of user

## How should this be manually tested?

- Start [Vikings backend](https://github.com/atlp-rwanda/vikings-ec-bn.git) 
- Clone [repo](https://github.com/atlp-rwanda/vikings-ec-fe.git)
- Change into the directory vikings-ec-fe
- Checkout to the branch: feature/Display-Users-184759911
- Run the command `npm install` 
- Run the command `npm run start`
- Login as admin 
- Go to https://vikings-ec-fe-git-feature-display-users-1-67492e-andela-vikings.vercel.app/dashboard/users
   * In case you get directed login as admin
- Click one the three dot of user you want to make change on
- Click on `make admin` or `make seller` or `make buyer` to change the role of user
- Click on `change account status` to disable or enable account

## Screenshots
Screenshot 2023-04-20 at 13 50 54
Screenshot (displaying user) 2023-04-20 at 13 43 35
Screenshot (update role) 2023-04-20 at 13 44 38
Screenshot (change user status)2023-04-20 at 13 44 18

### What are the relevant pivotal tracker stories?
[#184759911](https://www.pivotaltracker.com/story/show/184759911)
[#184759912](https://www.pivotaltracker.com/story/show/184759912)

### CODE REVIEW

**Review by ndatumuremyi — APPROVED** [review]

**Review by irakozetony — COMMENTED** [review]
Remove all consoles
Update PR description to indicate steps for testing it

**Review by irakozetony — APPROVED** [review]

**Review by nshimiyejayd — APPROVED** [review]

File: src/components/DropChildren.js
Comment (ncutixavier) [inline src/components/DropChildren.js:1]: Please rename this file to be specific with what you're implementing

File: src/components/forms/users/Table.js [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+        );
+        showSuccessMessage(response.payload.message);
+      }
+    } catch (error) {
+      showErrorMessage(error.data.message)
+    }
+  };
+
+  if (loading) {
+    return <div>Loading users...</div>;
```
Comment (ncutixavier) [inline src/components/forms/users/Table.js]: Try to use spinner component already created

File: src/components/products/BuyerViewSingleProduct.js [hunk 2]
```diff
@@ -19,9 +19,12 @@ const BuyerViewProduct = ({ ...props }) => {
...
   useEffect(() => {
     if (Array.isArray(ratings) && ratings.length > 0) {
-      const reviewers = ratings.map((r, index) => r.buyer);
+      const reviewers = ratings.map((rate) => ({
+        buyer: rate.buyer,
+        feedback: rate.feedback
+      }));
       setReviews(reviewers);
-      setIReviews(true);
+      setIsReviews(true);
```
Comment (ncutixavier) [inline src/components/products/BuyerViewSingleProduct.js]: Why did you make these changes?

File: src/features/api/customAxios.js [hunk 3]
```diff
@@ -16,7 +16,7 @@ const responseHandler = (response) => response;
 const errorHandler = (error) => {
   if (error.response.status === 401) {
     localStorage.clear();
-    return (location.href = `${process.env.FRONT_END_URL}/auth/signin`);
+    return (location.href = ` https://vikings-ec-fe-git-feature-display-users-1-67492e-andela-vikings.vercel.app/auth/signin`);
```
Comment (ncutixavier) [inline src/features/api/customAxios.js]: Why did you make these changes?

**Review by ncutixavier — CHANGES_REQUESTED** [review]
Please disable your formatter because you're doing unnecessary changes

File: src/components/DropDown.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
+import React, { useEffect, useRef, useState } from 'react';
+import Transition from './Transition';
+
+const DropChildren = ({
```
Comment (ncutixavier) [inline src/components/DropDown.js]: Update the function name

File: src/components/forms/users/Table.js
Comment (ncutixavier) [inline src/components/forms/users/Table.js:1]: - This is users related component, not forms
- The file name and function name should be related to the implementation

File: tests/components/DropChildren.test.js
Comment (ncutixavier) [inline tests/components/DropChildren.test.js:1]: Naming has not changed

**Review by ncutixavier — COMMENTED** [review]
- Please rebase, and check the naming of your file and function if they are related to the implementation
- Resolve the conflicts

File: src/store.js [hunk 5]
```diff
@@ -61,6 +65,11 @@ const store = configureStore({
     cart: cartReducer,
     addToCart: addToCartReducer,
     removeFromCart: removeFromCartReducer,
+    googleAuth: googleAuthReducer,
+    message: welcomeSlice,
+    users: userSlice,
+    roles: rolesSlice,
+    changeStatus: changeUserStatusSlice,
```
Comment (ncutixavier) [inline src/store.js]: You import reducer, not slice. Those are reducers, not slices. Please rename that, and if you have any issues you can ask @Abdden

**Review by ncutixavier — APPROVED** [review]

**Pull request merged**